azure-kusto-data/azure/kusto/data/helpers.py

```diff
@@ -1,14 +1,37 @@
 import sys
-from typing import TYPE_CHECKING, Union
+from typing import TYPE_CHECKING, Union, Callable, Dict, Optional
+from functools import lru_cache
+import copy
 
 if TYPE_CHECKING:
-    import pandas
+    import pandas as pd
     from azure.kusto.data._models import KustoResultTable, KustoStreamingResultTable
 
+# Alias for dataframe_from_result_table converter type
+Converter = Optional[Dict[str, Union[str, Callable[[str, "pd.DataFrame"], "pd.Series"]]]]
+
+
+@lru_cache(maxsize=1, typed=False)
+def default_dict() -> Dict[str, Callable[[str, "pd.DataFrame"], "pd.Series"]]:
+    import pandas as pd
+
+    return {
+        "string": lambda col, df: df[col].astype(pd.StringDtype()) if hasattr(pd, "StringDType") else df[col],
+        "guid": lambda col, df: df[col],
+        "dynamic": lambda col, df: df[col],
+        "bool": lambda col, df: df[col].astype(bool),
+        "int": lambda col, df: df[col].astype(pd.Int32Dtype()),
+        "long": lambda col, df: df[col].astype(pd.Int64Dtype()),
+        "real": lambda col, df: parse_float(df, col),
+        "decimal": lambda col, df: parse_float(df, col),
+        "datetime": lambda col, df: parse_datetime(df, col),
+        "timespan": lambda col, df: df[col].apply(to_pd_timedelta),
+    }
+
 
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License
-def to_pandas_timedelta(raw_value: Union[int, float, str]) -> "pandas.Timedelta":
+def to_pd_timedelta(raw_value: Union[int, float, str]) -> "pd.Timedelta":
```

Comment:
This should be called parse_timedelta and be with the rest of the similar methods at the bottom